src/Snippet.tsx

```diff
@@ -19,4 +22,51 @@ function SnippetFrame(code: ISnippet) {
   );
 }
 
-export {ISnippet, SnippetFrame};
+interface ISnippetProps {
+  blocks: string[];
+  lang: string;
+}
+
+enum Highlight {
+  None = 1,
+  Good,
+  Bad,
+}
+
+function highlightCssClass(h: Highlight) {
+  return Highlight[h].toLowerCase();
+}
+
+interface ISnippetState {
+  highlight: Highlight[];
+}
+
+class Snippet extends React.Component<ISnippetProps, ISnippetState> {
+  constructor(props: ISnippetProps) {
+    super(props);
+    const hls = [];
+    for (const _ of props.blocks) {
+      hls.push(Highlight.None);
+    }
+    this.state = { highlight: hls }
+  }
+
+  public render() {
+    const rb = [];
+    const langclass = "language-" + this.props.lang;
+    const classes = langclass + " block";
+    for (let i = 0; i < this.props.blocks.length; i++) {
+      const blkclss = classes + " " + highlightCssClass(this.state.highlight[i]);
+      rb.push(<div className={blkclss}>
+        <code className={blkclss}>{this.props.blocks[i]}</code>
+      </div>)
```

Comment:
That's weird. There's no semicolon here?

Reply:
oops

Reply:
how did you beat tslint 😮